src/main/java/com/mongodb/kafka/connect/util/jmx/internal/AdditiveMetric.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.mongodb.kafka.connect.util.jmx.internal;
+
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Consumer;
+
+public class AdditiveMetric implements Metric {
```

Comment:
I don't full grok how this works, can we add some docs to explain how it works and where / when it should be used.

Reply:
Docs added. Perhaps the name `AdditiveMetric` isn't very clear, but I'm not sure what else to call this. The name metric itself might also be confusing - there are multiple values (MBean attributes) that arise out of a single measurement (`sample(long)`).